### ISSUE TITLE
<algorithm>: /analyze with ranges algorithms emits fatal error C1060: compiler is out of heap space

### ISSUE THREAD

**Issue by StephanTLavavej** [microsoft/STL#1030]
We've been encountering a recurring problem where testing the ranges algorithms with `/analyze` will fail with `fatal error C1060: compiler is out of heap space`.

We're going to temporarily disable `/analyze` in `tests/std/tests/concepts_matrix.lst` until we can investigate this and develop a better workaround in the test machinery (or possibly report a bug to the static analysis compiler team).

**Comment by AlexGuteniev** [comment]
I've seen this with VS2015.
Large project is `/analyze`d by processing `.cpp` files one by one, and it seems to leak memory on each. Specifically, I observed that precompiled header is never unmapped.
I ended up setting large enough swap file (switching to using x64 toolset even for 32-bit x86 was done already by that time)

**Comment by StephanTLavavej** [comment]
That might have a different root cause (as these tests are single-TU). If it still repros with VS 2019, it's probably worth reporting.

Configuring our CI system to use x64-hosted x86-targeting binaries might be a mitigation, but it would be invasive. Reporting the ranges/concepts interaction to the analyze team, and using a targeted workaround, is probably less disruptive.

**Comment by cbezault** [comment]
Yeah changing the CI so that we could use the x64/x86 cross-compiler for tests other than tests with the `/BE` flag would be a pain.
